charts/cortex/templates/ingester/ingester-statefulset.yaml

```diff
@@ -141,3 +142,162 @@ spec:
             {{- toYaml . | nindent 12 }}
           {{- end }}
 {{- end -}}
+{{- if .Values.topologyAware.enabled }}
+{{- range $availabilityCell := .Values.topologyAware.availabilityCells }}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ template "cortex-ingester.fullname" $ }}-{{ $availabilityCell }}
+  labels: {{ include "cortex-ingester.labels" $ | nindent 4 }}
+{{ $.Values.topologyAware.zoneLabel | nindent 4 }}: {{ $availabilityCell }}
+  annotations:
+    {{- toYaml $.Values.ingester.annotations | nindent 4 }}
+spec:
+  {{- if not $.Values.ingester.autoscaling.enabled }}
+  replicas: {{ $.Values.ingester.replicas }}
+  {{- end }}
+  selector:
+    matchLabels: {{ include "cortex-ingester.matchLabels" $ | nindent 6 }}
+{{ $.Values.topologyAware.zoneLabel | nindent 6 }}: {{ $availabilityCell }}
+  updateStrategy:
+    {{- toYaml $.Values.ingester.statefulStrategy | nindent 4 }}          
+  podManagementPolicy: "{{ $.Values.ingester.statefulSet.podManagementPolicy }}"
+  serviceName: cortex-ingester-{{ $availabilityCell }}
```

Comment:
The service is created with name: `{{ template "cortex-ingester.fullname" $ }}-{{ $availabilityCell }}` should be the same here. Actually it could make sense to wrap it in a template (should take `availabilityCell` as argument as well as `$`)